### DISCUSSION TITLE
bug: Format pattern for searching not working as expected

### DISCUSSION THREAD

**Discussion by JustinBacher** [folke/noice.nvim#825] [Q&A] [answered]
### Did you check docs and existing issues?

- [X] I have read all the noice.nvim docs
- [X] I have searched the existing issues of noice.nvim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.10.0

### Operating system/version

Arch Linux

### Describe the bug

I assume using the default setup that noice would show an icon for search/replace but it isn't.

`/`
![image](https://github.com/folke/noice.nvim/assets/32350542/e6a1bbec-0c1a-4692-b6b0-091698975322)

`:` _(just entering the cmdline)_
![image](https://github.com/folke/noice.nvim/assets/32350542/9cf37376-a2d6-4f59-ae8e-3bdc4cc7c295)

`:%s/`
![image](https://github.com/folke/noice.nvim/assets/32350542/1cc22d32-cabb-4e29-aa04-9c79e21d15e6)

`:%s/abc`
![image](https://github.com/folke/noice.nvim/assets/32350542/0a7d37a5-e973-4406-b2c0-066bde0dee7f)

I have tried setting up with defaults and I have tried adding my own replace pattern. As in the seen in my reproducable example, I tested using `nvim -u repro.lua` and still not showing icon for replace so I don't think it's any of my plugins but I could be wrong

### Steps To Reproduce

1. Setup using default, or use custom replace formatting
2. Open cmdline
3. type: `%s/`

### Expected Behavior

For cmdline format to show the correct icon when entering search/replace

### Repro

```Lua
local root = vim.fn.fnamemodify("./.repro", ":p")

for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  {
    "folke/noice.nvim",
    dependencies = "MunifTanjim/nui.nvim",
    opts = {
      cmdline = {
        format = {
          replace = { pattern = "^:%s/", icon = "", lang = "vim" }
        }
      }
    }
  }
}

require("lazy").setup(plugins, { root = root .. "/plugins"})

vim.cmd.colorscheme("tokyonight")
```

**Comment by ilias777** [comment]
Hi. There is no icon for search and replace. The only available icons are described [here](https://github.com/folke/noice.nvim/wiki/What's-New%3F#cmdline-formatters). 

To search and replace you enter first the command line with `:` and for this is set the right arrow as icon. The *magnifying glass icon* is only for searching by pressing `/`.

**Reply by JustinBacher** [reply]
Is this not possible? Or is this just not implemented? Maybe I could PR this? If it's possible?

**Comment by ilias777** [comment]
I don't know if it's possible, because you entered first the command line with `:` and for this is the right arrow as icon. I think no.

**Reply by JustinBacher** [reply]
Right, but then how can `:lua ` be matched but `:%s/` can't be

**Reply by ilias777** [reply, marked as answer]
Yes, you are absolutely right. 
I played a little bit with the format options and with regex. And it works! Look here...


https://github.com/folke/noice.nvim/assets/39212564/033e2022-4d1a-42f4-8a53-06b572583d2e

Place this code inside the `opts` table:

```lua
cmdline = {
    format = {
        search_replace = { pattern = '^:%%s/', icon = '󰛔', lang = 'regex' },
    },
},
```

✌🏻🙂

**Reply by ilias777** [reply]
You can also add `kind = 'search',` before `pattern` and the border are changing the highlight and the text also changes from `Cmdline` to `Search`.

```lua
{ kind = 'search', pattern = '^:%%s/', icon = '󰛔', lang = 'regex' },

**Reply by JustinBacher** [reply]
Omg thank you so much! that's so simple it's been erking me for a bit now! didn't think it would be just adding the extra `%` Maybe this should be added to the recipes or the readme so people know!

**Reply by ilias777** [reply]
Glad that I helped you!
The extra `%` is to escape the `%s`, because in Lua Regex the `%s` is a whitespace.
Cool! Now you can mark this discussion as answered. 👍🏻
